framework/src/org/checkerframework/framework/util/typeinference/solver/EqualitiesSolver.java

```diff
@@ -381,6 +381,12 @@ public InferredValue mergeConstraints(
         dirty |= updateTargetsWithPartiallyInferredType(equalities, constraintMap, typeFactory);
         inferred = findEqualTarget(equalities, tops);
 
+        if (inferred == null && equalities.types.size() == 1) {
+            // Still could not find an inferred type in all hierarchies, so just use what type is
+            // known.
+            AnnotatedTypeMirror type = equalities.types.keySet().iterator().next();
+            return new InferredType(type);
```

Comment:
How about just setting inferred here and using the single return statement that follows.